File: python/solution_01.py
from utils import read_input

def part_1(input: str) -> int:
    floor: int = 0
    for c in input:
        if c == '(':
            floor += 1
        else:
            floor += -1
    return floor

def part_2(input: str) -> int:
    floor: int = 0
    count: int = 0
    for c in input:
        count += 1
        if c == '(':
            floor += 1
        else:
            floor += -1

        if floor < 0:
            return count

def solve() -> None:
    input = read_input("01").replace("\n", "")
    return part_1(input), part_2(input)


if __name__ == "__main__":
    floor, basement = solve()
    print(f"Floor: {floor}")
    print(f"Basement: {basement}")

File: rust/src/solutions/solution_01.rs
fn part_1(input: &str) -> i32{
    let mut floor: i32 = 0;
    for i in input.chars() {
        if i == '(' {
            floor += 1;
        }
        else {
            floor += -1;
        }
    }
    return floor;
}

fn part_2(input: &str) -> i32{
    let mut floor: i32 = 0;
    let mut count: i32 = 0;
    for i in input.chars() {
        count += 1;
        if i == '(' {
            floor += 1;
        }
        else {
            floor += -1;
        }

        if floor < 0 {
            return count;
        }
    }

    return 0;
}

pub fn solve(input: String) -> (i32, i32) {
    return (part_1(input.as_ref()), part_2(input.as_ref()));
}